### PR TITLE
Make proxy unsecure when adding policy without enabling security

### PR DESCRIPTION
**Pull request by ThishaniLucas**
Resolves the issues : https://wso2.org/jira/browse/ESBJAVA-4459 and https://github.com/wso2/product-ei/issues/1703

### CODE REVIEW

File: modules/core/src/main/java/org/apache/synapse/config/xml/ProxyServiceFactory.java [hunk 1]
```diff
@@ -259,52 +259,50 @@ public static ProxyService createProxy(OMElement elem, Properties properties) {
...
-                            pi.setType(PolicyInfo.MESSAGE_TYPE_IN);
-                        } else if ("out".equals(type.getAttributeValue())) {
-                            pi.setType(PolicyInfo.MESSAGE_TYPE_OUT);
-                        } else {
-                            handleException("Undefined policy type for the policy with key : "
-                                    + key.getAttributeValue());
+        if (isSecured(elem) && (getPolicies(elem) != null)) {
+            Iterator policies = getPolicies(elem);
+            while (policies.hasNext()) {
+                Object o = policies.next();
```
Comment (chanikag) [inline modules/core/src/main/java/org/apache/synapse/config/xml/ProxyServiceFactory.java]: Shall we have a meaningful name?

Comment (ThishaniLucas) [reply modules/core/src/main/java/org/apache/synapse/config/xml/ProxyServiceFactory.java]: Done

**Comment by lahirus** [comment]
"createProxy" method is too long and almost the whole class, which is not a problem with this PR. Better if we can break it meaningfully.

**Comment by lahirus** [comment]
There are no DEBUG logs in the class. Again it is not introduced from this PR, but better to add some DEBUG logs.

**Comment by ThishaniLucas** [comment]
Closing the issue since we need more testing.

**Pull request closed without merging**